src/honey/sql/util.cljc

```diff
@@ -76,3 +76,14 @@
 
       :default
       (clojure.string/join separator (transduce xform conj [] coll)))))
+
+(defn into
+  "Drop-in replacement for `clojure.core/into` that doesn't perform
+  transient-persistent roundtrip. Given the usage patterns and data sizes that
+  HoneySQL deals with, not going through transients is more efficient."
```

Comment:
I am slightly concerned that folks building large, complex SQL will be hitting thresholds where transients _are_ faster -- can you provide details on benchmarks for this, for both small and large SQL expressions (with 10s or maybe even 100s of parameters)?

Reply:
Well, you are right that it will be slower when there are 10s of parameters (the threshold is at about 10). It's not critical but should still be considered. Maybe, we can choose the better approach based on the number of elements in `from` – but I'm not a big fan of this approach.

Let me remove this commit from the PR for now and think about it more.